### DISCUSSION TITLE
How can I apply several locks to a model (without specifying the vehicle)?

### DISCUSSION THREAD

**Discussion by hklarner** [google/or-tools#3932] [Routing (and legacy CP) questions]
I have noticed that I cannot apply multiple locks to a problem. Only the most recent lock seems to be effective. Is this correct? How do I apply multiple locks without specifying the vehicle?

Below is a problem with two locks. Only the second lock (`[2, 4, 6]`) is effective in the assignment.

```python
n_stops = 10
n_vehicles = 1
depot = 0
index_manager = RoutingIndexManager(n_stops, n_vehicles, depot)
routing_model = RoutingModel(index_manager)

locks = [[9, 5, 7], [2, 4, 6]]
routing_model.ApplyLocks(locks=locks[0])
routing_model.ApplyLocks(locks=locks[1])

assignment = routing_model.Solve()  # nodes = [0, 2, 4, 6, 9, 8, 7, 5, 3, 1, 0]
```

**Comment by hklarner** [comment]
Another question: If I run this code with `n_vehicles = 2` then it crashes (!) with
```
Process finished with exit code 134 (interrupted by signal 6: SIGABRT)
```
Does this mean that the `ApplyLocks` call is meant only for models with one vehicle?

**Comment by Mizux** [comment]
Seems `ApplyLocks()` only  apply when vehicle number is 1 :shrug: 
https://github.com/google/or-tools/blob/1d696f9108a0ebfd99feb73b9211e2f5a6b0812b/ortools/constraint_solver/routing.cc#L2983-L2986

If I were you, I would try `ApplyLocksToAllVehicles()`
https://github.com/google/or-tools/blob/1d696f9108a0ebfd99feb73b9211e2f5a6b0812b/ortools/constraint_solver/routing.cc#L3004-L3008
